Installing extensions from the CLI fails on macOS

```
> oni2 -f --debug --trace --install-extension clangd
[DEBUG]   +72ms Oni2.Exception : Recording backtraces
[DEBUG]    +0ms Oni2.Exception : Recording backtraces
[DEBUG]    +0ms Oni2.Core.Setup : Looking for setup configuration at: /Applications/Onivim2.app/Contents/Resources/../MacOS/setup.json
[ERROR]    +0ms Service.Extensions.Management : Unable to install extension: clangd
[DEBUG]    +0ms Oni2.Core.ThreadHelper : Starting thread: 1 (Utility.waitForCondition)
[DEBUG]    +0ms Oni2.Core.ThreadHelper : Closing thread: 1 (Utility.waitForCondition)
Failed to install extension: clangd

> oni2 -f --debug --trace --install-extension exuberant-ctags
[DEBUG]   +45ms Oni2.Exception : Recording backtraces
[DEBUG]    +0ms Oni2.Exception : Recording backtraces
[DEBUG]    +0ms Oni2.Core.Setup : Looking for setup configuration at: /Applications/Onivim2.app/Contents/Resources/../MacOS/setup.json
[ERROR]    +0ms Service.Extensions.Management : Unable to install extension: exuberant-ctags
[DEBUG]    +0ms Oni2.Core.ThreadHelper : Starting thread: 1 (Utility.waitForCondition)
[DEBUG]    +0ms Oni2.Core.ThreadHelper : Closing thread: 1 (Utility.waitForCondition)
Failed to install extension: exuberant-ctags
```

Note that installing these extensions using the in-editor extensions UI works just fine.

--- a/scripts/lint.sh
+++ b/scripts/lint.sh
@@ -10,21 +10,21 @@ echo "prrerr_endline count: $PRERR_COUNT"
 echo "print_endline count: $PRINT_COUNT"
 echo "Printf.printf count: $PRINTF_COUNT"
 
-if [ "$PRERR_COUNT" -ne "6" ]; then
+if [ "$PRERR_COUNT" -ne "7" ]; then
     echo "New prerr_endline introduced; please remove and re-check:"
     echo "---"
     echo "Output: $(git grep prerr_endline)"
     exit 3
 fi
 
-if [ "$PRINT_COUNT" -ne "15" ]; then
+if [ "$PRINT_COUNT" -ne "16" ]; then
     echo "New print_endline introduced; please remove and re-check."
     echo "---"
     echo "Output: $(git grep print_endline)"
     exit 3
 fi
 
-if [ "$PRINTF_COUNT" -ne "2" ]; then
+if [ "$PRINTF_COUNT" -ne "3" ]; then
     echo "New Printf.printf introduced; please remove and re-check."
     echo "---"
     echo "Output: $(git grep Printf.printf)"

--- a/src/bin_editor/Cli.re
+++ b/src/bin_editor/Cli.re
@@ -1,0 +1,134 @@
+/*
+ * Cli.re
+ *
+ * Helpers for the editor command-line interface
+ */
+
+open Oni_CLI;
+
+module Core = Oni_Core;
+module ExtC = Service_Extensions.Catalog;
+module ExtM = Service_Extensions.Management;
+module Log = (val Core.Log.withNamespace("Oni2_editor"));
+module ReveryLog = (val Core.Log.withNamespace("Revery"));
+
+module LwtEx = Core.Utility.LwtEx;
+module OptionEx = Core.Utility.OptionEx;
+
+let installExtension = (path, Oni_CLI.{overriddenExtensionsDir, _}) => {
+  let setup = Core.Setup.init();
+  let result =
+    ExtM.install(~setup, ~extensionsFolder=?overriddenExtensionsDir, path)
+    |> LwtEx.sync;
+
+  switch (result) {
+  | Ok(_) =>
+    Printf.printf("Successfully installed extension: %s\n", path);
+    0;
+
+  | Error(_) =>
+    Printf.printf("Failed to install extension: %s\n", path);
+
+    let candidates: result(Service_Extensions.Catalog.SearchResponse.t, exn) =
+      ExtC.search(~offset=0, ~setup, path) |> LwtEx.sync;
+
+    switch (candidates) {
+    | Ok({extensions, _}) when extensions == [] => ()
+    | Ok({extensions, _}) =>
+      Printf.printf("\nDid you mean one of these extensions?\n");
+      extensions
+      |> List.iter((summary: ExtC.Summary.t) => {
+           print_endline(
+             Printf.sprintf("- %s.%s", summary.namespace, summary.name),
+           )
+         });
+      ();
+    | Error(exn) =>
+      prerr_endline("Error querying catalog: " ++ Printexc.to_string(exn))
+    };
+    1;
+  };
+};
+
+let uninstallExtension = (extensionId, {overriddenExtensionsDir, _}) => {
+  let result =
+    ExtM.uninstall(~extensionsFolder=?overriddenExtensionsDir, extensionId)
+    |> LwtEx.sync;
+
+  switch (result) {
+  | Ok(_) =>
+    Printf.sprintf("Successfully uninstalled extension: %s\n", extensionId)
+    |> print_endline;
+    0;
+
+  | Error(msg) =>
+    Printf.sprintf(
+      "Failed to uninstall extension: %s\n%s",
+      extensionId,
+      Printexc.to_string(msg),
+    )
+    |> prerr_endline;
+    1;
+  };
+};
+
+let printVersion = () => {
+  print_endline("Onivim 2 (" ++ Core.BuildInfo.version ++ ")");
+  0;
+};
+
+let queryExtension = (extension, _cli) => {
+  let setup = Core.Setup.init();
+  Service_Extensions.
+    // Try to parse the extension id - either search, or
+    // get details
+    (
+      switch (Catalog.Identifier.fromString(extension)) {
+      | Some(identifier) =>
+        Catalog.details(~setup, identifier)
+        |> LwtEx.sync
+        |> (
+          fun
+          | Ok(ext) => {
+              ext |> Catalog.Details.toString |> print_endline;
+              0;
+            }
+          | Error(msg) => {
+              prerr_endline(Printexc.to_string(msg));
+              1;
+            }
+        )
+      | None =>
+        Catalog.search(~offset=0, ~setup, extension)
+        |> LwtEx.sync
+        |> (
+          fun
+          | Ok(response) => {
+              response |> Catalog.SearchResponse.toString |> print_endline;
+              0;
+            }
+          | Error(msg) => {
+              prerr_endline(Printexc.to_string(msg));
+              1;
+            }
+        )
+      }
+    );
+};
+
+let listExtensions = ({overriddenExtensionsDir, _}) => {
+  Exthost.Extension.(
+    {
+      let extensions =
+        ExtM.get(~extensionsFolder=?overriddenExtensionsDir, ())
+        |> LwtEx.sync
+        |> Result.value(~default=[]);
+
+      let printExtension = (ext: Scanner.ScanResult.t) => {
+        print_endline(ext.manifest |> Manifest.identifier);
+      };
+      List.iter(printExtension, extensions);
+      0;
+    }
+  );
+};

--- a/src/bin_editor/Oni2_editor.re
+++ b/src/bin_editor/Oni2_editor.re
@@ -20,106 +20,6 @@ module ReveryLog = (val Core.Log.withNamespace("Revery"));
 module LwtEx = Core.Utility.LwtEx;
 module OptionEx = Core.Utility.OptionEx;
 
-let installExtension = (path, Oni_CLI.{overriddenExtensionsDir, _}) => {
-  let setup = Core.Setup.init();
-  let result =
-    ExtM.install(~setup, ~extensionsFolder=?overriddenExtensionsDir, path)
-    |> LwtEx.sync;
-
-  switch (result) {
-  | Ok(_) =>
-    Printf.printf("Successfully installed extension: %s\n", path);
-    0;
-
-  | Error(_) =>
-    Printf.printf("Failed to install extension: %s\n", path);
-    1;
-  };
-};
-
-let uninstallExtension = (extensionId, {overriddenExtensionsDir, _}) => {
-  let result =
-    ExtM.uninstall(~extensionsFolder=?overriddenExtensionsDir, extensionId)
-    |> LwtEx.sync;
-
-  switch (result) {
-  | Ok(_) =>
-    Printf.sprintf("Successfully uninstalled extension: %s\n", extensionId)
-    |> print_endline;
-    0;
-
-  | Error(msg) =>
-    Printf.sprintf(
-      "Failed to uninstall extension: %s\n%s",
-      extensionId,
-      Printexc.to_string(msg),
-    )
-    |> prerr_endline;
-    1;
-  };
-};
-
-let printVersion = () => {
-  print_endline("Onivim 2 (" ++ Core.BuildInfo.version ++ ")");
-  0;
-};
-
-let queryExtension = (extension, _cli) => {
-  let setup = Core.Setup.init();
-  Service_Extensions.
-    // Try to parse the extension id - either search, or
-    // get details
-    (
-      switch (Catalog.Identifier.fromString(extension)) {
-      | Some(identifier) =>
-        Catalog.details(~setup, identifier)
-        |> LwtEx.sync
-        |> (
-          fun
-          | Ok(ext) => {
-              ext |> Catalog.Details.toString |> print_endline;
-              0;
-            }
-          | Error(msg) => {
-              prerr_endline(Printexc.to_string(msg));
-              1;
-            }
-        )
-      | None =>
-        Catalog.search(~offset=0, ~setup, extension)
-        |> LwtEx.sync
-        |> (
-          fun
-          | Ok(response) => {
-              response |> Catalog.SearchResponse.toString |> print_endline;
-              0;
-            }
-          | Error(msg) => {
-              prerr_endline(Printexc.to_string(msg));
-              1;
-            }
-        )
-      }
-    );
-};
-
-let listExtensions = ({overriddenExtensionsDir, _}) => {
-  Exthost.Extension.(
-    {
-      let extensions =
-        ExtM.get(~extensionsFolder=?overriddenExtensionsDir, ())
-        |> LwtEx.sync
-        |> Result.value(~default=[]);
-
-      let printExtension = (ext: Scanner.ScanResult.t) => {
-        print_endline(ext.manifest |> Manifest.identifier);
-      };
-      List.iter(printExtension, extensions);
-      0;
-    }
-  );
-};
-
 Log.debug("Startup: Parsing CLI options");
 let (cliOptions, eff) = Oni_CLI.parse(~getenv=Sys.getenv_opt, Sys.argv);
 
@@ -128,12 +28,13 @@ if (cliOptions.needsConsole) {
 };
 
 switch (eff) {
-| PrintVersion => printVersion() |> exit
-| InstallExtension(name) => installExtension(name, cliOptions) |> exit
-| QueryExtension(name) => queryExtension(name, cliOptions) |> exit
-| UninstallExtension(name) => uninstallExtension(name, cliOptions) |> exit
+| PrintVersion => Cli.printVersion() |> exit
+| InstallExtension(name) => Cli.installExtension(name, cliOptions) |> exit
+| QueryExtension(name) => Cli.queryExtension(name, cliOptions) |> exit
+| UninstallExtension(name) =>
+  Cli.uninstallExtension(name, cliOptions) |> exit
 | CheckHealth => HealthCheck.run(~checks=All, cliOptions) |> exit
-| ListExtensions => listExtensions(cliOptions) |> exit
+| ListExtensions => Cli.listExtensions(cliOptions) |> exit
 | StartSyntaxServer({parentPid, namedPipe}) =>
   Oni_Syntax_Server.start(~parentPid, ~namedPipe, ~healthCheck=() =>
     HealthCheck.run(~checks=Common, cliOptions)

